src/actions/pkg.cr

```diff
@@ -1,45 +1,152 @@
-module DppmRestApi::Actions::Pkg
-  extend self
-  # List built packages
-  relative_get nil do |context|
-    if context.current_user? && Actions.has_access? context, Access::Read
-      # TODO: list all built packages
-      next context
-    end
-    raise Unauthorized.new context
-  end
-  # Clean unused built packages
-  relative_delete "/clean" do |context|
-    if context.current_user? && Actions.has_access? context, Access::Delete
-      # TODO: delete all unused built packages
-      next context
+require "../ext/semantic_version"
+
+module DppmRestApi::Actions
+  module Pkg
+    extend self
+    ALL_PKGS = ""
+    ONE_PKG  = "/:id"
+    include RouteHelpers
+
+    def build_config_response(package, builder)
+      builder.field package.package do
+        builder.object do
+          package.each_config_key do |key|
+            builder.field key, package.get_config key
+          end
+        end
+      end
     end
-    raise Unauthorized.new context
-  end
-  # Query information about a given package
-  relative_get "/:id/query" do |context|
-    if context.current_user? && Actions.has_access? context, Access::Read
-      # TODO: Query information about the given package
-      next context
+
+    def clean_unused_packages(context : HTTP::Server::Context)
+      if result = DppmRestApi.prefix.clean_unused_packages(confirmation: false) { }
+        return {data: result} if result.any?
+        bug = InternalServerError.new context, "received empty set from Prefix#clean_unused_packages; please report this strange bug"
+        error = NoPkgsToClean.new context
+        error.cause = bug
+        raise error
+      end
+      raise NoPkgsToClean.new context
     end
-    raise Unauthorized.new context
-  end
-  # Delete a given package
-  relative_delete "/:id/delete" do |context|
-    if context.current_user? && Actions.has_access? context, Access::Delete
-      # TODO: Query information about the given package
-      next context
+
+    struct ListResponse
+      include JSON::Serializable
+      property version : SemanticVersion
```

Comment:
All code relative to `SemanticVersion` isn't needed.
Also, what the `ListResponse` abstraction bring?
We will need similiar objects too for `app` and `src`.
The difference between the three will be:
- src: nothing, what's in pkgfile.
- pkg: add version
- app: add name (and perhaps version)

Reply:
It's not an abstraction, it's a type. We could use a NamedTuple if you're against having a defined struct...tbh I'm kind of confused as to what the issue is. The app and src versions can be `Actions::App::ListResponse` and `Actions::Src::ListResponse`.

Reply:
Does JSON.build directly wouldn't be better? In the sense of being flexible for what fields we return to the client, like description, info, etc.
This will avoid having 3 types, and the logic can be mostly shared.

Reply:
But why? What makes JSON.build better? It would be more code, and more prone to mistakes.

Reply:
How do you do to build only fields requested?

Reply:
Oh. That's a good point. If you want to do that, then yes, a `build` implementation would be more appropriate. If we are allowing bulk querying of config values, then I propose renaming this method to `/dppmrestapi/pkg/query`, and offer optional pagination on the response as a long-term goal once the project is more developed.

Reply:
... that's a good idea, I may implement this in the CLI at first.
Ok, we could let this as is and add a `TODO` comment.

Reply:
What config values were you thinking of allowing in the parameters for this? I didn't see anything relevant in `Pkg` or `Base`, unless you were planning to allow querying arbitrary config values in this route?

Reply:
Mainly what's in `PkgFille`, plus version for package and name for app.
This will help when we will build the Web client, listing all apps will require only one request.